types/storybook-react-router/index.d.ts

```diff
@@ -2,18 +2,18 @@
 // Project: https://github.com/gvaldambrini/storybook-router
 // Definitions by: Alex Lomia <https://github.com/alexlomm>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.1
 
-import * as React from 'react';
+import { ComponentType } from 'react';
 import { MemoryRouterProps } from 'react-router';
-import { Story, StoryDecorator } from '@storybook/react';
+import { DecoratorFunction, StoryApi } from '@storybook/addons';
 
-export const StoryRouter: React.ComponentType<{
-  story: Story;
+export const StoryRouter: ComponentType<{
+  story: StoryApi;
   links: object;
   routerProps: MemoryRouterProps;
 }>;
 
-declare function storyRouterDecorator(links?: object, routerProps?: MemoryRouterProps): StoryDecorator;
+declare function storyRouterDecorator(links?: object, routerProps?: MemoryRouterProps): DecoratorFunction;
```

Comment:
`DecoratorFunction` is a generic class and here we should use `DecoratorFunction<StoryFnReactReturnType>`. However `StoryFnReactReturnType` is not exported from `@storybook/react` so I think we can use `DecoratorFunction<any>` for now.

Reply:
StoryFnReactReturnType = ReactElement<unknown>

Reply:
I can import it from `'@storybook/react/dist/client/preview/types'`

```ts
import { StoryFnReactReturnType } from '@storybook/react/dist/client/preview/types';
```